### PR TITLE
[CP] Add document support for the API

### PR DESCRIPTION
**Pull request by chashikajw**
## Purpose
This will implement the Document related resources in backoffice DS and devportal DS.

## Fixes
- https://github.com/wso2/apk/issues/106

### CODE REVIEW

File: backoffice/backoffice-domain-service/ballerina/Dependencies.toml [hunk 1]
```diff
@@ -64,6 +64,23 @@ dependencies = [
 	{org = "ballerina", name = "time"}
 ]
 
+[[package]]
```
Comment (CrowleyRajapakse) [inline backoffice/backoffice-domain-service/ballerina/Dependencies.toml:67]: let's add same changes for template file

File: devportal/devportal-domain-service/ballerina/Dependencies.toml [hunk 2]
```diff
@@ -121,7 +121,7 @@ modules = [
 [[package]]
 org = "ballerina"
 name = "io"
-version = "1.4.0"
+version = "1.4.1"
```
Comment (CrowleyRajapakse) [inline devportal/devportal-domain-service/ballerina/Dependencies.toml:124]: add the same change for template file

**Comment by CrowleyRajapakse** [comment]
is document content searching not handled in this pr scope? if not let's create a separate issue to track that?

File: backoffice/backoffice-domain-service/ballerina/Dependencies.toml (hunk 1 above)
Comment (chashikajw) [reply backoffice/backoffice-domain-service/ballerina/Dependencies.toml:67]: Added from da7e3b8283a1e3da884958a332c7da14467b9f14

File: devportal/devportal-domain-service/ballerina/Dependencies.toml (hunk 2 above)
Comment (chashikajw) [reply devportal/devportal-domain-service/ballerina/Dependencies.toml:124]: Added from da7e3b8283a1e3da884958a332c7da14467b9f14

**Review by CrowleyRajapakse — APPROVED** [review]

**Pull request merged**